crates/db-arch/src/x86_64/cpuid/brand_string.rs

```
@@ -149,10 +147,14 @@ impl BrandString {
     /// `leaf` must be between 0x80000002 and 0x80000004.
     #[inline]
     pub fn get_reg_for_leaf(&self, leaf: u32, reg: Reg) -> u32 {
-        // It's ok not to validate parameters here, leaf and reg should
-        // both be compile-time constants. If there's something wrong with them,
-        // that's a programming error and we should panic anyway.
-        self.reg_buf[(leaf - 0x8000_0002) as usize * 4 + reg as usize]
+        if (0x80000002u32..=0x80000004).contains(&leaf) {
+            // It's ok not to validate parameters here, leaf and reg should
+            // both be compile-time constants. If there's something wrong with them,
+            // that's a programming error and we should panic anyway.
+            self.reg_buf[(leaf - 0x8000_0002) as usize * 4 + reg as usize]
+        } else {
+            0
```

"0" here could mean the value in the register is zero.
Should we return an error here to be more explicit?

---

we should return the default value for non-existing leaf functions